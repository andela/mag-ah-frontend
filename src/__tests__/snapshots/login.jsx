import React from "react";
import { MemoryRouter } from "react-router-dom";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import shallowToJSON from "enzyme-to-json";
import LoginPage from "../../components/login";

Enzyme.configure({ adapter: new Adapter() });

describe("The loginPage component", () => {
  const testProps = {
    dispatch: () => 1,
    error: "An error occurred",
    isAuthenticated: false
  };
  it("Should initially render without an error", () => {
    const tree = shallow(
      <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
        <LoginPage {...testProps} />
      </MemoryRouter>
    );
    expect(shallowToJSON(tree)).toMatchSnapshot();
  });
  it("Should show the error message when request fails", () => {
    const tree = shallow(
      <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
        <LoginPage {...testProps} />
      </MemoryRouter>
    );
    expect(shallowToJSON(tree)).toMatchSnapshot();
  });
});
