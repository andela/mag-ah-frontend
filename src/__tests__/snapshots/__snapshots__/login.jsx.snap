// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The loginPage component Should initially render without an error 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(LoginPage)
    dispatch={[Function]}
    error="An error occurred"
    isAuthenticated={false}
  />
</Router>
`;

exports[`The loginPage component Should show the error message when request fails 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(LoginPage)
    dispatch={[Function]}
    error="An error occurred"
    isAuthenticated={false}
  />
</Router>
`;
